cmds/exp/pxeserver/main.go

```diff
@@ -142,6 +144,7 @@ func main() {
 			mac:          maca,
 			self:         net.ParseIP(*selfIP),
 			yourIP:       yourIP,
+			subnet:        yourIP.Mask(yourNet.Mask),
```

Comment:
There's a space here in front of yourIP. is this gofmted?

Reply:
Isn't this gonna be 192.168.0.0, not the bcast addr?

Reply:
yeah, seems it wants that, seems to work just fine.

Reply:
Does that still work on Linux tho?

Reply:
it never worked for me on linux but I'll test. That use of subnet actually makes some sense, since we set a broadcast mac IIUC and the subnet is there to say "this is for you" to the interface.

Reply:
Just tested, works perfectly.

Reply:
2012/01/01 10:45:59 Got DHCPv4 lease on eth0: DHCPv4 Message
  opcode: BootReply
  hwtype: Ethernet
  hopcount: 0
  transaction ID: 0x1e1ba22f
  num seconds: 0
  flags: Unicast (0x00)
  client IP: 0.0.0.0
  your IP: 192.168.0.2
  server IP: 192.168.0.1
  gateway IP: 0.0.0.0
  client MAC: 00:07:32:4b:f1:38
  server hostname: 
  bootfile name: pxelinux.0
  options:
    Subnet Mask: ffffff00
    Router: 192.168.0.1
    IP Addresses Lease Time: 1193046h28m15s
    DHCP Message Type: ACK
    Server Identifier: 192.168.0.1

Reply:
and, whoa!!! the tftp server works fine.
I'm going to try it again with the fuctory bios and see if it works there too. If so, we can skip installing dhcpd and tftpd, oh happy day.

As part of the class we'll talk about http download ...